x-pack/plugins/security_solution/public/detection_engine/rule_exceptions/logic/use_exception_flyout_data.tsx

```diff
@@ -97,6 +97,24 @@ export const useFetchIndexPatterns = (rules: Rule[] | null): ReturnUseFetchExcep
       if (activeSpaceId !== '' && memoDataViewId) {
         setDataViewLoading(true);
         const dv = await data.dataViews.get(memoDataViewId);
+
+        // the autosuggest api fails if the .alerts index does not exist.
+        // So we need to append '*' char at end of .alerts-security.alerts-{spaceId}
+        // to prevent this from happening.
+        const newPattern = dv
+          .getIndexPattern()
+          .split(',')
+          .reduce((acc: string[], item: string) => {
+            if (item.includes('.alerts')) {
+              acc.push(item.concat(`*`));
+            } else {
+              acc.push(item);
+            }
+            return acc;
+          }, [] as string[])
```

Comment:
I _think_ you can get rid of the `as` cast here. When I removed it it continued to type it correctly.